### PR TITLE
fixed factory for unsigned transactions

### PR DESCRIPTION
**Pull request by iulianpascalau**
fixed factory for unsigned transactions
added single shard integration tests for unsigned transactions

### CODE REVIEW

File: integrationTests/singleShard/unsignedTransaction/testInitializer.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
+package unsignedTransaction
```
Comment (sasurobert) [inline integrationTests/singleShard/unsignedTransaction/testInitializer.go]: no need for this file. The 2 tests could stay in the same package with transactions.

File: p2p/libp2p/netMessenger.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ const DirectSendID = protocol.ID("/directsend/1.0.0")
 const refreshPeersOnTopic = time.Second * 60
 const ttlPeersOnTopic = time.Second * 120
 
+var messageHeader = 200
```
Comment (sasurobert) [inline p2p/libp2p/netMessenger.go]: is this number always the same ?

File: integrationTests/singleShard/unsignedTransaction/interceptedBulkTx_test.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	"github.com/ElrondNetwork/elrond-go/data/state/addressConverters"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNode_GenerateSendInterceptBulkUnsignedTransactionsWithMessenger(t *testing.T) {
```
Comment (sasurobert) [inline integrationTests/singleShard/unsignedTransaction/interceptedBulkTx_test.go]: this is similar with transaction interceptor test, too much duplicated code. Do the test with adding the differences as parameters. A set of integration tests could be made for every request TOPIC.

File: integrationTests/singleShard/unsignedTransaction/testInitializer.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/unsignedTransaction/testInitializer.go]: done

File: p2p/libp2p/netMessenger.go (hunk 2 above)
Comment (iulianpascalau) [reply p2p/libp2p/netMessenger.go]: Not quite. Changed to a higher value (64kB) just to be sure and addes a TODO to remove this message size when https://github.com/libp2p/go-libp2p-pubsub/pull/189 will be part of a new release.

File: integrationTests/singleShard/unsignedTransaction/interceptedBulkTx_test.go (hunk 3 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/unsignedTransaction/interceptedBulkTx_test.go]: Will try to extract more reusable methods

Comment (iulianpascalau) [reply integrationTests/singleShard/unsignedTransaction/interceptedBulkTx_test.go]: Done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNode_GenerateSendInterceptBulkUnsignedTransactionsWithMessenger(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	dPool := createTestDataPool()
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	startingNonce := uint64(6)
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	addrConverter, _ := addressConverters.NewPlainAddressConverter(32, "0x")
+	accntAdapter := createAccountsDB()
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+	shardCoordinator := &sharding.OneShardCoordinator{}
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+	time.Sleep(time.Second)
+
+	wg := sync.WaitGroup{}
+	wg.Add(noOfUnsignedTx)
+
+	chanDone := make(chan bool)
+
+	go func() {
+		wg.Wait()
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+		mut.Lock()
+		defer mut.Unlock()
+
+		unsignedtxHashes = append(unsignedtxHashes, key)
+
+		dataStore := dPool.UnsignedTransactions().ShardDataStore(
+			process.ShardCacherIdentifier(shardCoordinator.SelfId(), shardCoordinator.SelfId()),
+		)
+		val, _ := dataStore.Get(key)
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+		Data:    "tx notarized data",
+		TxHash:  []byte("tx hash"),
+	}
+
+	scrBuff, _ := testMarshalizer.Marshal(scr)
+
+	fmt.Printf("Unsigned transaction: %v\n%v\n", scr, string(scrBuff))
+
+	chanDone := make(chan bool)
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+		RcvAddr: testHasher.Compute("receiver"),
+		SndAddr: buffPk1,
+		Data:    "tx notarized data",
+		TxHash:  []byte("tx hash"),
+	}
+
+	scrBuff, _ := testMarshalizer.Marshal(scr)
+
+	fmt.Printf("Unsigned transaction: %v\n%v\n", scr, string(scrBuff))
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+		Nonce:   0,
+		Value:   big.NewInt(0),
+		RcvAddr: testHasher.Compute("receiver"),
+		SndAddr: buffPk1,
+		Data:    "tx notarized data",
+		TxHash:  []byte("tx hash"),
+	}
+
+	scrBuff, _ := testMarshalizer.Marshal(scr)
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: delete empty line

File: integrationTests/singleShard/transaction/testInitializer.go [hunk 13]
```diff
@@ -268,3 +263,71 @@ func getConnectableAddress(mes p2p.Messenger) string {
...
+		assert.Fail(t, fmt.Sprintf("should have been %d, got %d", noOfTxs, len(txHashes)))
+
+		return
+	}
+
+	bitmap := make([]bool, noOfTxs+int(startingNonce))
+	//set for each nonce from found tx a true flag in bitmap
+	for i := 0; i < noOfTxs; i++ {
+		val, _ := cache.ShardDataStore(
+			process.ShardCacherIdentifier(shardCoordinator.SelfId(), shardCoordinator.SelfId()),
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/testInitializer.go]: move this into a variable.

File: integrationTests/singleShard/transaction/testInitializer.go [hunk 14]
```diff
@@ -268,3 +263,71 @@ func getConnectableAddress(mes p2p.Messenger) string {
...
+		).Get(txHashes[i])
+
```
Comment (sasurobert) [inline integrationTests/singleShard/transaction/testInitializer.go]: delete empty line

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 4 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 5 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 6 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 7 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 8 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go (hunk 9 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go (hunk 10 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go (hunk 11 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go (hunk 12 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedResolvedUnsignedTx_test.go]: done

File: integrationTests/singleShard/transaction/testInitializer.go (hunk 13 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/testInitializer.go]: done

File: integrationTests/singleShard/transaction/testInitializer.go (hunk 14 above)
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/testInitializer.go]: done

**Pull request merged**